### ISSUE TITLE
trailing slashes on directories return no results

### ISSUE THREAD

**Issue by rijnhard** [mrmlnc/fast-glob#290]
### Environment

  * OS Version: Linux Mint 20 5.4.0-47-generic
  * Node.js Version: 12.18.4

### Actual behavior
With settings
```json5
{
  onlyFiles      : false,
  onlyDirectories: true,
  markDirectories: true,
  absolute     : true,
  globstar      : false
}
```
Pattern `test/fixtures/*/` returns no results.
But Pattern `test/fixtures/*` does return results.

### Expected behavior
Directory Patterns `test/fixtures/*/` and `test/fixtures/*/` should be equivalent.

### Code sample

```js
import fglob from 'fast-glob';

const dirGlobSettings =  {
  onlyFiles      : false,
  onlyDirectories: true,
  markDirectories: true,
  absolute     : true,
  globstar      : false
};

const relativeDirGlob1 = 'test/fixtures/*/';
const relativeDirGlob2 = 'test/fixtures/*';

fglob.sync(relativeDirGlob1, dirGlobSettings);
fglob.sync(relativeDirGlob2, dirGlobSettings);
```

Side note: should the output should respect `markDirectories`?

**Comment by Trott** [comment]
I ran `git bisect` to find where this bug was introduced and it's in 5ddccbc3951489e6476ccb2177a9c54ba915981b. That's a really small change set that was a refactor. No bug fix or new feature. So I suspect reverting it will fix this bug.

**Comment by Trott** [comment]
Proposed fix (or at least a fix for most cases) in https://github.com/mrmlnc/fast-glob/pull/335.

**Comment by mrmlnc** [comment]
Fixed in the `master` branch by #339. The release with `3.2.8` version will be released soon.
